Upgrading from 1.13 to 1.14 leaves triggers in broken state

<!-- 
Are you using Knative? If you do, we would love to know!
https://github.com/knative/community/issues/new?template=ADOPTERS.yaml&title=%5BADOPTERS%5D%3A+%24%7BCOMPANY+NAME+HERE%7D
-->

**Describe the bug**
Upgrading from `knative=1.13.8` and `kafka-extensions=1.13.11`.

knative-eventing related resources update and restart as expected.
eventing-kafka extension related resource upgrade fails:

1. `kafka-broker-dispatcher` statefulset set starts at 3 replicas and eventually scales up to 15 replicas in our dev environment. Only 3/15 replicas end up with data populated to the corresponding `kafka-broker-dispatcher-X` configmap resource. I see a continued process of the pods going through container creating state as it waits for the CM to get created and then throws an error when it finally starts up
```
{"@timestamp":"2024-08-30T21:12:19.847Z","@version":"1","message":"failed to parse from JSON","logger_name":"dev.knative.eventing.kafka.broker.core.eventbus.ContractPublisher","thread_name":"contract-file-watcher","level":"ERROR","level_value":40000,"stack_trace":"com.google.protobuf.InvalidProtocolBufferException: Expect message object but got: null\n\tat com.google.protobuf.util.JsonFormat$ParserImpl.mergeMessage(JsonFormat.java:1481)\n\tat com.google.protobuf.util.JsonFormat$ParserImpl.merge(JsonFormat.java:1458)\n\tat com.google.protobuf.util.JsonFormat$ParserImpl.merge(JsonFormat.java:1322)\n\tat com.google.protobuf.util.JsonFormat$Parser.merge(JsonFormat.java:486)\n\tat dev.knative.eventing.kafka.broker.core.eventbus.ContractPublisher.parseFromJson(ContractPublisher.java:98)\n\tat dev.knative.eventing.kafka.broker.core.eventbus.ContractPublisher.updateContract(ContractPublisher.java:74)\n\tat dev.knative.eventing.kafka.broker.dispatcher.main.Main.lambda$start$0(Main.java:137)\n\tat dev.knative.eventing.kafka.broker.core.file.FileWatcher.run(FileWatcher.java:124)\n\tat java.base/java.lang.Thread.run(Thread.java:1595)\n"}
```
2. The `kafka-controller-post-install` job completes and outputs no logs.

If (2) is replaced with a manual step to cleanup the old dispatcher deployment, all event traffic ceases and triggers go into a failed state:
```
dev020-ke.a.a.v1.training-complete               knative-wa-a-broker                     107d   False   Schedule
dev020-ke.a.a.v1.training-failed                 knative-wa-a-broker                     107d   False   Schedule
dev020-ke.a.a.v2.training-start                  knative-wa-a-broker                     462d   False   Schedule
dev020-ke.a.c-c.v1.training-complete          knative-wa-c-broker                           386d   False   Schedule
dev020-ke.a.c-c.v1.training-failed            knative-wa-c-broker                           386d   False   PropagateSubscriberURI
```
Recreation of any trigger does not help. Existing broker resources continue to report ready but if any are recreated will also go into a failed state. A describe on any trigger shows a failure on old/no-longer-existing dispatcher deployment
```
  - lastTransitionTime: "2024-08-30T20:23:29Z"
    message: 'failed to schedule consumers: failed to get data plane volume kafka-resources
      in pod knative-eventing/kafka-broker-dispatcher-6b6855d6c5-7nhtw'
    reason: Schedule
    status: "False"
    type: Ready
```

 **Expected behavior**

 **To Reproduce**
Steps to reproduce the behavior.

 **Knative release version**
`knative=1.13.8` and `kafka-extensions=1.13.11`

 **Additional context**
Add any other context about the problem here such as proposed priority

I deployed 1.14.12 with sarama debug logging turned on and observed the error when triggers go down after working for a while:

```
[sarama][debug] 2024/09/26 18:09:47 github.com/IBM/sarama@v1.43.1/client.go:1027: client/metadata fetching metadata for all topics from broker kafka-0:9093
[sarama][debug] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:1027: client/metadata fetching metadata for all topics from broker kafka-2:9093
[sarama] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:1073: client/metadata got error from broker 2 while fetching metadata: write tcp x.x.x.x:52230->x.x.x.x:9093: write: broken pipe
[sarama] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/broker.go:344: Error while closing connection to broker kafka-2:9093: tls: failed to send closeNotify alert (but connection was closed anyway): write tcp x.x.x.x:52230->x.x.x.x:9093: write: broken pipe
[sarama] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:774: client/brokers deregistered broker #2 at kafka-2:9093
[sarama][debug] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:1027: client/metadata fetching metadata for all topics from broker kafka-0:9093
[sarama] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:1073: client/metadata got error from broker 0 while fetching metadata: EOF
[sarama][debug] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/broker.go:342: Closed connection to broker kafka-0:9093
[sarama] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:774: client/brokers deregistered broker #0 at kafka-0:9093
[sarama][debug] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:1027: client/metadata fetching metadata for all topics from broker kafka-1:9093
[sarama] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:1073: client/metadata got error from broker 1 while fetching metadata: write tcp x.x.x.x:51990->x.x.x.x:9093: write: broken pipe
[sarama] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/broker.go:344: Error while closing connection to broker kafka-1:9093: tls: failed to send closeNotify alert (but connection was closed anyway): write tcp x.x.x.x:51990->x.x.x.x:9093: write: broken pipe
[sarama] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:774: client/brokers deregistered broker #1 at kafka-1:9093
[sarama][debug] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:1027: client/metadata fetching metadata for all topics from broker kafka-0:9093
[sarama] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/client.go:1073: client/metadata got error from broker -1 while fetching metadata: write tcp x.x.x.x:41544->x.x.x.x:9093: write: broken pipe
[sarama] 2024/09/26 18:19:47 github.com/IBM/sarama@v1.43.1/broker.go:344: Error while closing connection to broker kafka-0:9093: tls: failed to send closeNotify alert (but connection was closed anyway): write tcp x.x.x.x:41544->x.x.x.x:9093: write: broken pipe
```

which seems to indicate Knative/Sarama client is not properly closing/refreshing connections.

I have also observed the same `EOF` error but on start up of kafka-controller pods. We run with 3 replicas and during this error, when the pod come up at least 1 out of the 3 pods will immediately hit the `EOF` errors. This brings the triggers down and as far as I have seen the pod never recovers. The other 2 pods are up and communicate to kafka fine. The impacted pod has full network connectivity to the kafka instance. Most importantly all 3 pods have the same kafka config... so 2/3 connecting with one failing SASL handshake is supicious.

```
[sarama] 2024/09/26 18:58:52 github.com/IBM/sarama@v1.43.1/config.go:618: ClientID is the default of 'sarama', you should consider setting it to something application-specific.
[sarama][debug] 2024/09/26 18:58:52 github.com/IBM/sarama@v1.43.1/client.go:1027: client/metadata fetching metadata for all topics from broker kafka-1:9093
[sarama] 2024/09/26 18:58:52 github.com/IBM/sarama@v1.43.1/broker.go:1250: Error while performing SASL handshake kafka-1:9093
[sarama][debug] 2024/09/26 18:58:52 github.com/IBM/sarama@v1.43.1/broker.go:267: Closed connection to broker kafka-1:9093
[sarama] 2024/09/26 18:58:52 github.com/IBM/sarama@v1.43.1/client.go:1073: client/metadata got error from broker -1 while fetching metadata: EOF
[sarama] 2024/09/26 18:58:52 github.com/IBM/sarama@v1.43.1/client.go:1086: client/metadata no available broker to send metadata request to
[sarama] 2024/09/26 18:58:52 github.com/IBM/sarama@v1.43.1/client.go:788: client/brokers resurrecting 3 dead seed brokers
```

I also tried with 
```
      ENABLE_SARAMA_LOGGER:                       true
      ENABLE_SARAMA_DEBUG_LOGGER:                 true
      ENABLE_SARAMA_CLIENT_POOL:                  false
```
which resulted in the latter example of pods coming up with immediate EOF errors and down eventing.

Yes. I have tried with `ENABLE_SARAMA_CLIENT_POOL=false`, and I see the same problems. 

Over the weekend I left the test cluster on 1.14 w/ client pool disabled and all triggers are currently marked as Not Ready/InitializeOffset. 1/3 kafka-controller pods is throwing constant `EOF` errors while the other 2 pods have no errors. Despite triggers reporting not ready, I am able to send events to them and they get routed to our receivers. Example error logs:
```
[sarama] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/client.go:1012: client/metadata retrying after 250ms... (0 attempts remaining)
[sarama] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/config.go:618: ClientID is the default of 'sarama', you should consider setting it to something application-specific.
[sarama][debug] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/client.go:1027: client/metadata fetching metadata for all topics from broker kafka-2:9093
[sarama] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/broker.go:1250: Error while performing SASL handshake kafka-2:9093
[sarama][debug] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/broker.go:267: Closed connection to broker kafka-2:9093
[sarama] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/client.go:1073: client/metadata got error from broker -1 while fetching metadata: EOF
[sarama] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/config.go:618: ClientID is the default of 'sarama', you should consider setting it to something application-specific.
[sarama][debug] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/client.go:1027: client/metadata fetching metadata for all topics from broker kafka-0:9093
[sarama] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/broker.go:1250: Error while performing SASL handshake kafka-0:9093
[sarama][debug] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/broker.go:267: Closed connection to broker kafka-0:9093
[sarama] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/client.go:1073: client/metadata got error from broker -1 while fetching metadata: EOF
[sarama] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/config.go:618: ClientID is the default of 'sarama', you should consider setting it to something application-specific.
[sarama][debug] 2024/09/30 16:37:29 github.com/IBM/sarama@v1.43.1/client.go:1027: client/metadata fetching metadata for all topics from broker kafka-1:9093
[sarama] 2024/09/30 16:37:30 github.com/IBM/sarama@v1.43.1/broker.go:1250: Error while performing SASL handshake kafka-1:9093
[sarama][debug] 2024/09/30 16:37:30 github.com/IBM/sarama@v1.43.1/broker.go:267: Closed connection to broker kafka-1:9093
[sarama] 2024/09/30 16:37:30 github.com/IBM/sarama@v1.43.1/client.go:1073: client/metadata got error from broker -1 while fetching metadata: EOF
[sarama] 2024/09/30 16:37:30 github.com/IBM/sarama@v1.43.1/client.go:1086: client/metadata no available broker to send metadata request to
[sarama] 2024/09/30 16:37:30 github.com/IBM/sarama@v1.43.1/client.go:788: client/brokers resurrecting 3 dead seed brokers
[sarama][debug] 2024/09/30 16:37:30 github.com/IBM/sarama@v1.43.1/client.go:309: Closing Client
```

--- a/control-plane/pkg/reconciler/consumergroup/auth.go
+++ b/control-plane/pkg/reconciler/consumergroup/auth.go
@@ -27,32 +27,27 @@ import (
 )
 
 func (r *Reconciler) newAuthSecret(ctx context.Context, cg *kafkainternals.ConsumerGroup) (*corev1.Secret, error) {
-	var secret *corev1.Secret
-
 	if hasSecretSpecConfig(cg.Spec.Template.Spec.Auth) {
 		secret, err := security.Secret(ctx, &SecretSpecSecretLocator{cg}, security.DefaultSecretProviderFunc(r.SecretLister, r.KubeClient))
 		if err != nil {
 			return nil, err
 		}
+		return secret, nil
+	}
 
-		authContext, err := security.ResolveAuthContextFromLegacySecret(secret)
-		if err != nil {
-			return nil, err
-		}
-		return authContext.VirtualSecret, nil
-
-	} else if hasNetSpecAuthConfig(cg.Spec.Template.Spec.Auth) {
+	if hasNetSpecAuthConfig(cg.Spec.Template.Spec.Auth) {
 		auth, err := security.ResolveAuthContextFromNetSpec(r.SecretLister, cg.GetNamespace(), *cg.Spec.Template.Spec.Auth.NetSpec)
 		if err != nil {
 			return nil, err
 		}
-		secret, err = security.Secret(ctx, &NetSpecSecretLocator{cg}, security.NetSpecSecretProviderFunc(auth))
+		secret, err := security.Secret(ctx, &NetSpecSecretLocator{cg}, security.NetSpecSecretProviderFunc(auth))
 		if err != nil {
 			return nil, fmt.Errorf("failed to get secret: %w", err)
 		}
+		return secret, nil
 	}
 
-	return secret, nil
+	return nil, nil
 }
 
 type NetSpecSecretLocator struct {

--- a/control-plane/pkg/reconciler/consumergroup/consumergroup.go
+++ b/control-plane/pkg/reconciler/consumergroup/consumergroup.go
@@ -293,14 +293,14 @@ func (r *Reconciler) reconcileStatusSelector(cg *kafkainternals.ConsumerGroup) {
 }
 
 func (r *Reconciler) deleteConsumerGroupMetadata(ctx context.Context, cg *kafkainternals.ConsumerGroup) error {
-	kafakSecret, err := r.newAuthSecret(ctx, cg)
+	kafkaSecret, err := r.newAuthSecret(ctx, cg)
 	if err != nil {
 		return fmt.Errorf("failed to get secret for Kafka cluster auth: %w", err)
 	}
 
 	bootstrapServers := kafka.BootstrapServersArray(cg.Spec.Template.Spec.Configs.Configs["bootstrap.servers"])
 
-	kafkaClusterAdminClient, err := r.GetKafkaClusterAdmin(ctx, bootstrapServers, kafakSecret)
+	kafkaClusterAdminClient, err := r.GetKafkaClusterAdmin(ctx, bootstrapServers, kafkaSecret)
 	if err != nil {
 		return fmt.Errorf("cannot obtain Kafka cluster admin, %w", err)
 	}

